File: contracts/models/PlaybackQueue.kt
package contracts.models

import contracts.enums.RepeatMode
import kotlinx.serialization.Serializable

/**
 * Single item inside a playback queue, referencing logical Media ID.
 */
@Serializable
data class PlaybackItem(
    val mediaId: String,            // Logical Media ID
    val title: String,
    val durationMs: Long,
    val thumbnailUri: String,
    val artistOrChannel: String,
    val preferredSourceId: String? = null
)

/**
 * Universal playback queue state.
 */
@Serializable
data class PlaybackQueue(
    val queueId: String,            // que_01j7b...
    val currentIndex: Int = 0,
    val items: List<PlaybackItem> = emptyList(),
    val repeatMode: RepeatMode = RepeatMode.OFF,
    val isShuffled: Boolean = false
)

File: contracts/models/PlaybackQueue.ts
import { RepeatMode } from "../enums/RepeatMode";

export interface PlaybackItem {
  mediaId: string;
  title: string;
  durationMs: number;
  thumbnailUri: string;
  artistOrChannel: string;
  preferredSourceId?: string | null;
}

export interface PlaybackQueue {
  queueId: string;
  currentIndex: number;
  items: PlaybackItem[];
  repeatMode: RepeatMode;
  isShuffled: boolean;
}
